### DISCUSSION TITLE
Suggestion: A CTRF to static HTML output

### DISCUSSION THREAD

**Discussion by timkrins** [ctrf-io/ctrf#10] [Reporters]
Again, fantastic idea, and excited to see current activity on this project!

One very popular thing that most testing frameworks contain is the ability to output their tests into a static HTML file for deployment or saving as a CI run artefact.

It would be great if we can take a CTRF file and produce a static HTML report.
(even better if it includes some client-side logic to group or filter tests)

Some examples for some frameworks:
- [mochawesome-report-generator](https://github.com/adamgruber/mochawesome-report-generator)
- [jest-html-reporters](https://github.com/Hazyzh/jest-html-reporters)
- [jest-stare](https://github.com/dkelosky/jest-stare)
- [pytest-html](https://github.com/pytest-dev/pytest-html)

CTRF is on its way to being some very awesome glue for self-run CI pipelines!

I am keen myself to create something - I might get some time to do this over the next month or so.

**Comment by timkrins** [comment]
I did some quick research on some recent static site generators - and whipped this up very very quickly using Astro.

It uses the example CTRF at https://ctrf.io/docs/schema/examples#comprehensive-report

<img width="969" alt="image" src="https://github.com/ctrf-io/ctrf/assets/531798/602d1ed7-298a-490c-85f6-8eeadd9f313b">

I can't spend any more time on it right now, but I'm keen to put together a much more advanced version of this.

**Reply by Ma11hewThomas** [reply]
Great idea! I am looking forward to seeing what you build

**Comment by Ma11hewThomas** [comment]
#26 regarding timelines

**Comment by garaichinar-cyber** [comment]
How about using existing tools : for example, convert CTRF format to Allure format and use Allure as visualisation tool and as generator of static HTML ? 

Allure docs : https://allurereport.org/docs/how-it-works/

**Comment by agracio** [comment]
I know that this is an old thread but if you have CTRF JUnit output then https://github.com/agracio/mochawesome-converter can convert it to Mochawesome report.
If there is a demand for it I will add a native CTRF converter.

**Comment by redben** [comment]
FWIW, this seems to do the job without intermediate format https://github.com/ryu-sato/ctrf-html-reporter/blob/main/README.md

**Comment by agracio** [comment]
https://github.com/agracio/mochawesome-converter now has native support for CTRF JSON format.
